wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java

```diff
@@ -1207,13 +1208,13 @@ private byte[] parseAsHexOrBase58(String data) {
         }
     }
 
-    private long getCreationTimeSeconds() {
+    private Optional<Instant> getCreationTime() {
         if (unixtime != null)
-            return unixtime;
+            return Optional.of(Instant.ofEpochSecond(unixtime));
         else if (date != null)
-            return date.getTime() / 1000;
+            return Optional.of(date.toInstant());
         else
-            return 0;
+            return Optional.empty();
```

Comment:
```
    return (unixtime != null)
                    ? Optional.of(Instant.ofEpochSecond(unixtime))
                    : Optional.ofNullable(date.toInstant());
```

Reply:
or
```
return Optional.ofNullable(unixtime != null
                              ? Instant.ofEpochSecond(unixtime)
                              :  date != null
                                          ? date.toInstant()
                                          : null);
 ```

Reply:
I see your smiley (-:

Honestly, without a reason (for example being forced to use only functions in a constructor) I prefer the if-else-else. And in any case, I would change that in a followup commit so that this one stays as small and reviewable as possible.